src/kOS/Execution/TransferManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using kOS.Safe;
+using kOS.Suffixed;
+
+namespace kOS.Execution
+{
+    public class TransferManager : IUpdateObserver
+    {
+        public enum TransferStatus        
+        {
+            Failed,
+            Finished,
+            Inactive,
+            Transferring
+        }
+
+        private readonly SharedObjects shared;
+        private readonly HashSet<ResourceTransferValue> transfers;
+
+        public TransferManager(SharedObjects shared)
+        {
+            this.shared = shared;
+            transfers = new HashSet<ResourceTransferValue>();
+            shared.UpdateHandler.AddFixedObserver(this);
+        }
+
+        public ResourceTransferValue CreateTransfer(PartResourceDefinition resourceInfo, object transferTo, object transferFrom, double amount)
+        {
+            var toReturn = new ResourceTransferValue(this, resourceInfo, transferTo, transferFrom, amount);
+            transfers.Add(toReturn);
+            return toReturn;
+        }
+
+        public ResourceTransferValue CreateTransfer(PartResourceDefinition resourceInfo, object transferTo, object transferFrom)
+        {
+            var toReturn = new ResourceTransferValue(this, resourceInfo, transferTo, transferFrom);
+            transfers.Add(toReturn);
+            return toReturn;
+        }
+
+        public void Dispose()
+        {
+            shared.UpdateHandler.RemoveFixedObserver(this);
```

Comment:
yeah, the manager there is only one. The reason each transfer doesnt get "disposed" is that the user can reactivate one after it is finished. i imagined this working on a TRANSFERALL command and you just want to keep nudging it.